Gems/Atom/RHI/Code/Include/Atom/RHI/ShaderResourceGroupData.h

```diff
@@ -241,21 +232,27 @@ namespace AZ
             uint32_t GetUpdateMask() const;
             
             //! Update the indirect buffer view with the indices of all the image views which reside in the global gpu heap.
+            //! Ideally higher level code can access bindless heap indices directly from the view and populate any indirect
+            //! buffer directly. This API is present in case we want RHI to track bindless resources which may be needed for
+            //! backends like the metal.
             void SetBindlessViews(
                 ShaderInputBufferIndex indirectResourceBufferIndex,
                 const RHI::BufferView* indirectResourceBuffer,
                 AZStd::span<const ImageView* const> imageViews,
                 uint32_t* outIndices,
-                bool viewReadOnly = true,
+                AZStd::span<bool> isViewReadOnly,
                 uint32_t arrayIndex = 0);
```

Comment:
is this a breaking API change?  If so we should probably bump the Atom RPI major version in `gem.json` to `1.0.0` and if you think it might affect many users, consider leaving a deprecated override version of the old function.